test/inductor/test_indexing.py

```diff
@@ -10,6 +12,84 @@
 from torch.utils._sympy.functions import FloorDiv, ModularIndexing
 
 
+class TestIndirectIndexing(TorchTestCase):
+    def test_indirect_indexing_oob_crashes(self):
+        fn1 = """
+@torch.compile
+def fn(x):
+    b = torch.arange(3)
+    return x[b]
+
+fn(torch.zeros(2))
+"""
+
+        fn2 = """
+@torch.compile
+def fn(x):
+    b = torch.tensor([-3])
+    return x[b]
+
+fn(torch.zeros(2))
+"""
+        fn3 = """
+@torch.compile
+def fn(x):
+    b = torch.tensor([3])
+    return x[b]
+
+fn(torch.zeros(2))
+"""
+        fn4 = """
+@torch.compile
+def fn(x):
+    b = torch.tensor([0])
+    return x[b + 3]
+
+fn(torch.zeros(2))
+"""
+        for fn in [fn1, fn2, fn3, fn4]:
+            with self.assertRaises(
+                subprocess.CalledProcessError,
+            ):
+                try:
+                    # Run the deserialized function in a subprocess
+                    result = subprocess.run(
+                        [
+                            "python",
+                            "-c",
+                            f"import torch\n{fn}",
+                        ],
+                        capture_output=True,
+                        text=True,
+                        check=True,
+                    )
+
+                except subprocess.CalledProcessError as e:
+                    self.assertEqual(e.returncode, -6)
+                    self.assertTrue("index out of bounds" in e.stderr)
+                    raise
+
+    def test_indirect_indexing_negative_index_in_bounds(self):
+        fn1 = """
```

Comment:
This test doesn't need subprocess as it ought not to crash the process

Reply:
@jon-chuang, I'd like to include these tests in https://github.com/pytorch/pytorch/pull/113068. Would you be OK if I just copy the diff into the other PR?

Reply:
Hello @guilhermeleobas sure, go ahead :)